src/aura/AuraAvatarMultiToken.sol

```diff
@@ -572,26 +562,37 @@ contract AuraAvatarTwoToken is BaseAvatar, PausableUpgradeable, AuraAvatarUtils,
     // INTERNAL
     ////////////////////////////////////////////////////////////////////////////
 
-    /// @notice Unstakes the given amount of assets and transfers them back to owner.
-    /// @dev This function doesn't claim any rewards. Caller can only be owner.
-    /// @param _amountAsset1 Amount of asset1 to be unstaked.
-    /// @param _amountAsset2 Amount of asset2 to be unstaked.
-    function _withdraw(uint256 _amountAsset1, uint256 _amountAsset2) internal {
-        if (_amountAsset1 == 0 && _amountAsset2 == 0) {
-            revert NothingToWithdraw();
+    function _addBptPositionInfo(uint256 _newPid) internal {
+        if (pids.contains(_newPid)) {
+            revert PidAlreadyExist(_newPid);
         }
+        pids.add(_newPid);
+        (address lpToken,,, address crvRewards,,) = AURA_BOOSTER.poolInfo(_newPid);
+        assets.add(lpToken);
+        baseRewardPools.add(crvRewards);
+        // Boster approval for bpts
+        IERC20MetadataUpgradeable(lpToken).safeApprove(address(AURA_BOOSTER), type(uint256).max);
+    }
 
-        if (_amountAsset1 > 0) {
-            baseRewardPool1.withdrawAndUnwrap(_amountAsset1, false);
-            asset1.safeTransfer(msg.sender, _amountAsset1);
+    /// @notice Unstakes the given amount of assets and transfers them back to owner.
+    /// @dev This function doesn't claim any rewards. Caller can only be owner.
+    /// @param _pids PIDs to be targetted to unstake from.
+    /// @param _amountAssets Amount of assets to be unstaked.
+    function _withdraw(uint256[] memory _pids, uint256[] memory _amountAssets) internal {
+        for (uint256 i; i < _pids.length;) {
+            if (_amountAssets[i] == 0) {
+                revert NothingToWithdraw();
+            }
+            // TODO: Cache
+            (address lpToken,,, address crvRewards,,) = AURA_BOOSTER.poolInfo(_pids[i]);
 
-            emit Withdraw(address(asset1), _amountAsset1, block.timestamp);
-        }
-        if (_amountAsset2 > 0) {
-            baseRewardPool2.withdrawAndUnwrap(_amountAsset2, false);
-            asset2.safeTransfer(msg.sender, _amountAsset2);
+            IBaseRewardPool(crvRewards).withdrawAndUnwrap(_amountAssets[i], false);
```

Comment:
Why not claiming rewards along the way? As it stands, should we withdraw with any outstanding rewards, these will be stuck there unless we call `claimRewardsAndSendToOwner()` which will require manual Msig work to process. If we were to claim them, anything outstanding would be transferred to the Avatar and process during the following harvest (or atomically should harvest be called manually).

Maybe I am missing something..

Reply:
it was a decision to keep separated the actions of deposit, wd, rewards claiming etc. As it stands now, even in the case that owner wds funds from specific position, as long as the sets contains the info around the base reward pool contract in storage, when the CL keeper will trigger internally `claimAndRegisterRewards` method, which will loop thru all the rewards avail, will claim them and process them as usual

Reply:
I see, perfect. I just confirmed that the `removeBptPositionInfo()` function will check if there are any rewards pending and claim them before removing the info, that would have been my only concern. Good stuff!
```
        // NOTE: verify pending rewards and claim. Processing is done separately
        if (baseRewardPool.earned(address(this)) > 0) {
            baseRewardPool.getReward();
        }
```